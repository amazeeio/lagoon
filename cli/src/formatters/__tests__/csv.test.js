// @flow

import formatAsCsv from '../csv';

describe('csv formatter', () => {
  it('should format data with 0 data rows correctly', () => {
    const formatted = formatAsCsv([['Name', 'Customer']]);
    expect(formatted).toMatchSnapshot();
  });

  it('should format data with 1 data row correctly', () => {
    const formatted = formatAsCsv([
      ['Name', 'Customer'],
      ['test-project-csv-1', 'test-customer-csv-1'],
    ]);
    expect(formatted).toMatchSnapshot();
  });

  it('should format data with > 1 data rows correctly', () => {
    const formatted = formatAsCsv([
      ['Name', 'Customer'],
      ['test-project-csv-1', 'test-customer-csv-1'],
      ['test-project-csv-2', 'test-customer-csv-2'],
    ]);
    expect(formatted).toMatchSnapshot();
  });
});
