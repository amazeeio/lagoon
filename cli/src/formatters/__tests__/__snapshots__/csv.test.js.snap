// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`csv formatter should format data with > 1 data rows correctly 1`] = `
"Name,Customer
test-project-csv-1,test-customer-csv-1
test-project-csv-2,test-customer-csv-2
"
`;

exports[`csv formatter should format data with 0 data rows correctly 1`] = `
"Name,Customer
"
`;

exports[`csv formatter should format data with 1 data row correctly 1`] = `
"Name,Customer
test-project-csv-1,test-customer-csv-1
"
`;
